watcher/types.go

```diff
@@ -85,6 +85,28 @@ func (v *BackendWatcher) watch() {
 				continue
 			}
 
+			var addresses []v1.EndpointAddress
+			for _, a := range endpoint.Subsets[0].Addresses {
+				puid := string(a.TargetRef.UID)
+
+				po, err := v.client.CoreV1().Pods(v.namespace).List(metav1.ListOptions{
+					FieldSelector: fields.OneTermEqualSelector("metadata.uid", puid).String(),
+				})
+
+				if err != nil {
+					glog.Errorf("error while locating endpoint : %s", err.Error())
+					continue
+				}
+
+				if po.Items[0].Status.Conditions[0].Status != v1.ConditionTrue {
```

Comment:
Not enough paranoia:

```suggestion
				if len(po.Items) > 0 && len(po.Items[0].Status.Conditions) > 0 && po.Items[0].Status.Conditions[0].Status != v1.ConditionTrue {
```